### PR TITLE
Improve torch.library.opcheck and register_autograd docs

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141883
* #141880
* #141894

Fixes https://github.com/pytorch/pytorch/issues/141618

### CODE REVIEW

File: torch/library.py [hunk 1]
```diff
@@ -1321,12 +1325,12 @@ def opcheck(
 
         >>> # xdoctest: +REQUIRES(env:TORCH_DOCTEST_CUDA)
         >>> @torch.library.custom_op("mylib::numpy_mul", mutates_args=())
-        >>> def numpy_add(x: Tensor, y: float) -> Tensor:
+        >>> def numpy_mul(x: Tensor, y: float) -> Tensor:
```
Comment (albanD) [inline torch/library.py:1328]: The op is actually and addition, not a mul? :p

Comment (zou3519) [reply torch/library.py:1328]: whoops

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**